librz/core/cdebug.c

```diff
@@ -885,24 +885,31 @@ static void print_debug_maps_ascii_art(RzDebug *dbg, RzList *maps, ut64 addr, in
 }
 
 RZ_API void rz_debug_map_list_visual(RzDebug *dbg, ut64 addr, const char *input, int colors) {
-	if (dbg) {
-		int i;
-		for (i = 0; i < 2; i++) { // Iterate over dbg::maps and dbg::maps_user
-			RzList *maps = (i == 0) ? dbg->maps : dbg->maps_user;
-			if (maps) {
-				RzListIter *iter;
-				RzDebugMap *map;
-				if (input[1] == '.') { // "dm=." Only show map overlapping current offset
-					dbg->cb_printf("TODO:\n");
-					rz_list_foreach (maps, iter, map) {
-						if (addr >= map->addr && addr < map->addr_end) {
-							// print_debug_map_ascii_art (dbg, map);
-						}
-					}
-				} else { // "dm=" Show all maps with a graph
-					print_debug_maps_ascii_art(dbg, maps, addr, colors);
+	if (!dbg) {
+		return;
+	}
+	int i;
+	for (i = 0; i < 2; i++) { // Iterate over dbg::maps and dbg::maps_user
+		//		RzList *maps = (i == 0) ? dbg->maps : dbg->maps_user;
+		RzList *maps = rz_debug_map_list(dbg, (bool)i);
+		if (!maps) {
+			continue;
+		}
+		RzListIter *iter;
+		RzDebugMap *map;
+		if (input[1] == '.') { // "dm=." Only show map overlapping current offset
+			dbg->cb_printf("TODO:\n");
+			rz_list_foreach (maps, iter, map) {
+				if (addr >= map->addr && addr < map->addr_end) {
+					// print_debug_map_ascii_art (dbg, map);
 				}
 			}
+		} else { // "dm=" Show all maps with a graph
+			print_debug_maps_ascii_art(dbg, maps, addr, colors);
 		}
 	}
 }
+
+RZ_API RzList *rz_debug_map_list(RzDebug *dbg, bool user_map) {
```

Comment:
I recommend to move this function to `librz/debug/dmap.c` instead, it doesn't need to be in the core.
Please also add a short Doxygen description.